src/main/java/org/opensearch/knn/jni/JNIUtils.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.jni;
+
+import com.sun.jna.Platform;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import oshi.util.platform.mac.SysctlUtil;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+import java.util.Locale;
+
+public class JNIUtils {
+
+    private static final Logger logger = LogManager.getLogger(JNIUtils.class);
+
+    /**
+     * Verify if the underlying system supports AVX2 SIMD Optimization or not
+     * 1. If the architecture is aarch64 return false.
+     * 2. If the operating system is windows return false.
+     * 3. If the operating system is macOS, use oshi-core library to verify if the cpu flags
+     *    contains 'avx2' and return true if it exists else false.
+     * 4. If the operating system is linux, read the '/proc/cpuinfo' file path and verify if
+     *    the flags contains 'avx2' and return true if it exists else false.
+     */
+    public static boolean isAVX2SupportedBySystem() {
+        if (Platform.isARM() || Platform.isWindows()) {
```

Comment:
If we are checking OS later on, we dont need to check windows here.

also, can we switch to:
```
if (!Platform.isX86()) {
  return false;
}
```

This will be more explicit so that when we support other archs, like PowerPC, we wont need to change this code.

Reply:
Makes sense